operationsgateway_api/src/records/waveform.py

```diff
@@ -79,19 +84,38 @@ def _create_plot(self, buffer) -> None:
         plt.clf()
 
     @staticmethod
-    async def get_waveform(waveform_id: str) -> WaveformModel:
+    def get_relative_path(record_id: str, channel_name: str) -> str:
         """
-        Given a waveform ID, find the waveform that's stored in MongoDB. This function
-        assumes that the waveform should exist; if no waveform can be found, a
-        `MissingDocumentError` will be raised
+        Returns a relative waveform path given a record ID and channel name. The path is
+        relative to the base directory of where waveforms are stored in Echo
         """
-        waveform_data = await MongoDBInterface.find_one(
-            "waveforms",
-            {"_id": waveform_id},
-        )
+        return f"{record_id}/{channel_name}.json"
+
+    @staticmethod
+    def get_full_path(relative_path: str) -> str:
+        """
+        Converts a relative waveform path to a full path by adding the 'prefix' onto a
+        relative path of a waveform. The full path doesn't include the bucket name
+        """
+        return f"{Waveform.echo_prefix}/{relative_path}"
 
-        if waveform_data:
+    @staticmethod
+    async def get_waveform(waveform_path: str) -> WaveformModel:
```

Comment:
Does get_waveform need to be asynchronus?

Reply:
Fixed in 861ebc2e4d86e1d6bd48ef1eba48cb77c4a98c7d